account_invoice_split_refund/wizard/sale_make_invoice_advance.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models, fields, api
+
+
+class SaleAdvancePaymentInv(models.TransientModel):
+
+    _inherit = "sale.advance.payment.inv"
+
+    advance_payment_method = fields.Selection(
+        selection_add=[
+            ('all_split', 'Invoiceable lines (Split refunds from invoices)')
+        ],
+        default=lambda self: self._get_advance_payment_method(),
+    )
+
+    @api.model
+    def _get_advance_payment_method(self):
+        """Force splitting refunds from invoices as default method"""
+        return 'all_split'
+
+    @api.multi
+    def create_invoices(self):
+        if self.advance_payment_method != 'all_split':
+            return super(SaleAdvancePaymentInv, self).create_invoices()
+        sale_orders = self.env['sale.order'].browse(
+            self._context.get('active_ids', []))
+        # Create all the invoices first
+        sale_orders.action_invoice_create()
+        # Create any refund afterwards
+        sale_orders.action_invoice_create(final=True)
```

Comment:
if this line fails it will affect result of 29?

Reply:
If it fails, the transaction is rolled back and that's perfectly fine.

Reply:
if first call successful, second fails, result of first rolled back, I am thinking that we twice calls same function and it raise Usererror

Reply:
It won't work if there is no invoice or no refund to create. I think it will work only if there is an invoice and refund to create. You need to add test condition or try/catch

Reply:
Fixed in 7bcf35d